### PR TITLE
Don't return bytes if py3 in salt-cloud

### PR DESCRIPTION
**Pull request by cachedout**
### What does this PR do?

We used to return bytes instead of unicode, which would make a dict with mixed bytes and unicode. This was a pain and caused tests to fail.
### What issues does this PR fix or reference?

Failing py3 compat in salt-cloud
### Previous Behavior

A dict that looked like `
..{'default': {'disk': 0, 'id': b'1', 'driver': <libcloud.compute.drivers.dimensiondata.DimensionDataNodeDriver object at 0x7f2fcbc769b0>, 'price': 0, 'get_uuid': <bound method UuidMixin.get_uuid of <NodeSize: id=1, name=default, ram=0 disk=0 bandwidth=0 price=0 driver=DimensionData ...>>, 'ram': 0, 'name': b'default', 'uuid': b'94811c79b526d5ef99518ce215d56123cc644067', 'bandwidth': 0, 'extra': {}}}
` for the test return in `unit.cloud.clouds.dimensiondata_test.DimensionDataTestCase.test_avail_sizes`

cc: @techhat @rallytime and @s0undt3ch 

@s0undt3ch This affects changes that you made specifically, so please have a quick look if you can.

### CODE REVIEW

**Comment by rallytime** [comment]
@cachedout Can you clean up those [pylint errors](https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/2710/violations/)?

**Comment by cachedout** [comment]
@rallytime Thanks. Fixed.

File: salt/cloud/libcloudfuncs.py [hunk 1]
```diff
@@ -274,17 +276,18 @@ def get_image(conn, vm_):
     '''
     images = conn.list_images()
 
-    vm_image = config.get_cloud_config_value('image', vm_, __opts__).encode(
-        'ascii', 'salt-cloud-force-ascii'
-    )
+    if not six.PY3:
+        vm_image = config.get_cloud_config_value('image', vm_, __opts__).encode(
```
Comment (rallytime) [inline salt/cloud/libcloudfuncs.py]: Don't we still need a `vm_image` set if we are running python 3? Similar to the change in line 246 above: https://github.com/saltstack/salt/pull/34105/files#diff-e8c33e1c4a43de8e11379a6d5cf064b0R246

Comment (cachedout) [inline salt/cloud/libcloudfuncs.py]: Gah. Yeah. Sorry. Thanks.

**Comment by s0undt3ch** [comment]
Looked. :+1:

**Pull request merged**